bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -5938,6 +6041,192 @@ private static void closeStatement(Statement statement) {
+	 *
+	 * @param ingestJobId The ingest job to set the end date for
+	 * @param endDateTime The end date
+	 *
+	 * @throws TskCoreException If inserting into the database fails
+	 */
+	void setIngestJobEndDateTime(long ingestJobId, long endDateTime) throws TskCoreException {
+		CaseDbConnection connection = connections.getConnection();
+		acquireSharedLock();
+		try {
+			Statement statement = connection.createStatement();
+			statement.executeUpdate("UPDATE ingest_jobs SET end_date_time=" + endDateTime + " WHERE ingest_job_id=" + ingestJobId + ";");
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error updating the end date (ingest_job_id = " + ingestJobId + ".", ex);
+		} finally {
+			connection.close();
+			releaseSharedLock();
+		}
+	}
+
+	void setIngestJobStatus(long ingestJobId, IngestJobStatusType status) throws TskCoreException {
+		CaseDbConnection connection = connections.getConnection();
+		acquireSharedLock();
+		try {
+			Statement statement = connection.createStatement();
+			statement.executeUpdate("UPDATE ingest_jobs SET status_id=" + status.ordinal() + " WHERE ingest_job_id=" + ingestJobId + ";");
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error ingest job status (ingest_job_id = " + ingestJobId + ".", ex);
+		} finally {
+			connection.close();
+			releaseSharedLock();
+		}
+	}
+
+	/**
+	 * Adds the given ingest job to the database
+	 *
+	 * @param dataSource    The datasource the ingest job is being run on
+	 * @param hostName      The name of the host
+	 * @param ingestModules The ingest modules being run during the ingest job.
+	 *                      Should be in pipeline order.
+	 * @param jobStart      The time the job started
+	 *
+	 * @return An information object representing the ingest job added to the
+	 *         database.
+	 *
+	 * @throws TskCoreException If adding the job to the database fails.
+	 */
+	public final IngestJobInfo addIngestJob(Content dataSource, String hostName, List<IngestModuleInfo> ingestModules, Date jobStart, Date jobEnd, IngestJobStatusType status, String settingsDir) throws TskCoreException {
+		CaseDbConnection connection = connections.getConnection();
+		acquireSharedLock();
+		ResultSet resultSet = null;
+		Statement statement = null;
+		try {
+			statement = connection.createStatement();
+			String update = "INSERT INTO ingest_jobs (data_src_id, host_name, start_date_time, end_date_time, status_id, settings_dir) "
+					+ "VALUES (" + dataSource.getId() + ", '" + hostName + "', " + jobStart.getTime() + ", " + jobEnd.getTime() + ", " + status.ordinal() + ", '" + settingsDir + "');";
+			statement.executeUpdate(update);
+			resultSet = statement.getGeneratedKeys();
+			resultSet.next();
+			long id = resultSet.getLong(1);
+			for (int i = 0; i < ingestModules.size(); i++) {
+				IngestModuleInfo ingestModule = ingestModules.get(i);
+				statement.executeUpdate("INSERT INTO ingest_job_modules (ingest_job_id, ingest_module_id, pipeline_position) "
```

Comment:
Because ingest_jobs and ingest_job_modules are updated in this method and an error updating ingest_job_modules means that the data is incorrect, the database operations in this method should be done in a transaction.